scripts/install/jdownloader.sh

```diff
@@ -0,0 +1,117 @@
+    read -r 'myjd_email'
+    echo_query "Please enter the password for the account."
+    read -r 'myjd_password'
+    echo_query "Please enter the device name they would like this JDownloader installation to show up as."
+    read -r 'myjd_devicename'
+
+    # Install JDownloader
+    echo_progress_start "Downloading and installing JDownloader for $user"
+    mkdir -p /home/"$user"/jd
+    wget -q http://installer.jdownloader.org/JDownloader.jar -O /home/"$user"/jd/JDownloader.jar
+    # Run JDownloader once to generate the majority of files and dirs.
+    java -jar /home/"$user"/jd/JDownloader.jar -norestart >> ${log} 2>&1
+    # Check if JDownloader's first run was successful.
+    # TODO: Figure out if there is a better file or folder for this test, whichever file is generated last would be best.
+    if [[ -e "/home/$user/jd/cfg" ]]; then
+        echo_info "JDownloader's first run was likely successful."
+    else
+        echo_info "JDownloader's first run likely failed. Exiting."
+        exit 2
+    fi
+
+    # Pass my.jdownloader.org information to org.jdownloader.api.myjdownloader.MyJDownloaderSettings.json
+    echo_progress_start "Adding the users https://my.jdownloader.org/ information to their installation."
+    cat > /home/"$user"/jd/cfg/org.jdownloader.api.myjdownloader.MyJDownloaderSettings.json << EOF
+    {
+      "email" : "$myjd_email",
+      "password" : "$myjd_password",
+      "devicename" : "$myjd_devicename"
+    }
+EOF
+
+    systemctl enable -q --now jdownloader@"$user"
+}
+
+# If there was a variable passed to this script, it isn't the initial installation.
+# It is likely being called because a user is being added with "box adduser".
+# Install JDownloader for just this user, and exit.
+if [[ -n $1 ]]; then
+    user=$1
+    install_jdownloader "${user}"
+    exit 0
+fi
+
+# If we made it through the previous block. The script has likely been called from "box install".
+# Install Java
+# use "java -version" to check if it even needs installation
+STR=$(java --version)
+SUB='not found'
+if [[ "$STR" == *"$SUB"* ]]; then
+    echo_info "Java was not found. swizzin will need to install it."
+    echo_progress_start "Downloading and installing a Java runtime environment."
+    apt_update
+    apt_install default-jre
+    # use "java -version" to verify installation
+    STR=$(java --version)
+    SUB='not found'
+    if [[ "$STR" == *"$SUB"* ]]; then
+        echo_info "Java was not installed correctly. Check the swizzin log for details. Exiting."
+        exit 1
+    else
+        echo_info "Java was installed successfully."
+    fi
+else
+    echo_info "Java was found. No need to install."
+fi
```

Comment:
use the `java` functions we already have in `sources/functions/java`, I think they're merged in with airsonic

Reply:
We did discuss this in passing on the Discord. I believe we decided to just have me do my own thing? As the java scripts for airsonic were Java 8 specific for some reason or another. I don't think JDownloader will care either way, because as far as I know it only requires `Java 1.8`.

I'm open to discuss it further if you would like.

Reply:
I don't see a reason to deviate from JRE8 and using the wheel we already invented. Maintaining an extra install of JRE as we'll have to start maintaining those in all of the service files for Java apps.

Is there any outstanding advantage when using +8, which would outweigh the added complexity to the system?

Reply:
I think it would actually be nice to move some of the "testing" logic into the function defined here so that we use it anytime we install JDK

https://github.com/swizzin/swizzin/blob/master/sources/functions/java

this means we should handle the entire java install above hrough something like
```sh
. /etc/swizzin/sources/functions/java
install_jdk8
```

Reply:
This change seems fine to me. Like I said, I don't think JDownloader is going to are either way.

Except, I should note that I've moved away from:
```
STR=$(java --version)
SUB='not found'
if [[ "$STR" == *"$SUB"* ]]; then
```
because it was not functioning properly under certain conditions. I'm not sure I could replicate them but it was because `java --version` was outputting something other than "command not found" when Java was not installed, and therefore skipping installation. I think possibly the system had just recently had Java removed, but I could be mistaken.
I have been using this instead since. Haven't had any issues since. 
```
if [[ ! -e /usr/bin/java ]]; then
```
I'm not sure where you would want to put them into the `install_java8` function itself. I'm a little confused because it looks like this function only handles `buster` installs. I'm I reading this incorrectly?

Reply:
> I'm not sure where you would want to put this logic into the install_java8 function itself. I'm a little confused because it looks like this function only handles buster installs. Am I reading this incorrectly?

It handles buster explicitly, and all others (the `*)` case) are handled through the apt install.

We _might_ want to check for **pre-existing** JRE installs before the actual install here and try to exit gracefully or offer some mitigations.

Reply:
Ahhhhh see. I didn't know about `*)` case. This makes more sense now. If I remember correctly. You mentioned something about airsonic requiring Java 8 specifically for some reason or another. So for the install_java8 function, I'd recommend checking for `/usr/bin/java` first, and then check to see if a version other than 8 is installed. Then have some sort of user intervention at that point. They could choose to (A) Cancel. (B) Remove their current installation for a Java 8 install instead. (C) If this is somehow possible, opt into having parallel java installations. (All apps that _require_ `Java 8` would have to act accordingly)

Reply:
This is very rough, but essentially something like this.
```
if [[ ! -e /usr/bin/java ]]; then
    echo_info "Java is not installed. We are free to install Java 8"
else
    echo_info "Java is installed. Check which version"
    if [[ ! "$(/usr/bin/java --version)" == *'build 8.'* ]]; then
        echo_error "Current Java version is not 8."
        echo_info "We can handle this in three ways:"
        echo_info "(A) "
        echo_info "(B) "
        echo_info "(C) "
        echo_query "How do you want to handle this?"
        read -r "answer"
        case $answer in
            "A")
                echo_info "Do A"
                ;;
            "B")
                echo_info "Do B"
                ;;
            "C")
                echo_info "Do C"
                ;;
            *)
                echo_info "Loop back and get a proper answer."
                ;;
        esac
    fi
fi
```

Reply:
Hm.

Fuck it, that's for another PR. We should re-evaluate if it's mybe time to upgrade the JRE too, but that's out of scope for this app introduction. If we use the function we have for it now, then making the changes there later will apply to the whole toolset, and that will be slick.

Reply:
Sounds good to me. Will change it as soon as I have all the install issues sorted out

Reply:
The script now installs Java with the swizzin function.
What should we call this other PR for Java changes? Should I start it? or should you?

Reply:
I'd say `fix: java management` is good enough. Feel free to have a start at it ;)